### PR TITLE
DB-ADDON Plugin:

### PR DESCRIPTION
**Pull request by sisamiwe**
- Remove use of _plg_item_dict in WebIF
- automate generation of file item_attributes.py and update of plugin.yaml per script in item_attributes_master.py
- update docu
- bump to 1.1.2

### CODE REVIEW

File: db_addon/item_attributes_master.py [hunk 1]
```diff
@@ -151,9 +189,8 @@ def get_attrs(sub_dict: dict = {}) -> list:
                 attributes.append(db_addon_fct)
     return attributes
 
-
-def export_db_addon_data():
-    ATTRS = {}
+def export_item_attributs_py():
```
Comment (onkelandy) [inline db_addon/item_attributes_master.py:192]: There is a typo

File: db_addon/item_attributes_master.py [hunk 2]
```diff
@@ -168,27 +205,70 @@ def export_db_addon_data():
...
+
+        valid_list_calculation = f"""{valid_list_calculation}\n\
+          - '{ITEM_ATTRIBUTS[attribute][db_addon_fct]['calc']:<}'"""
+
+    valid_list_calculation = f"""{valid_list_calculation}\n\r"""
+
+    return valid_list_str, valid_list_desc_str, valid_list_item_type, valid_list_calculation
+
+def update_plugin_yaml_avm_data_type():
+    """Update ´'valid_list', 'valid_list_description', 'valid_list_item_type' and 'valid_list_calculation' of item attributes in plugin.yaml"""
```
Comment (onkelandy) [inline db_addon/item_attributes_master.py:246]: `' before valid_list

File: db_addon/item_attributes_master.py [hunk 3]
```diff
@@ -168,27 +205,70 @@ def export_db_addon_data():
...
-        print()
+    if data.get('item_attributes', {}).get(attribute):
+        data['item_attributes'][attribute]['valid_list'] = yaml.load(valid_list_str)
+        data['item_attributes'][attribute]['valid_list_description'] = yaml.load(valid_list_desc_str)
+        data['item_attributes'][attribute]['valid_list_item_type'] = yaml.load(valid_list_item_type_str)
+        data['item_attributes'][attribute]['valid_list_calculation'] = yaml.load(valid_list_calc_str)
 
+        with open(FILENAME_PLUGIN, 'w', encoding="utf-8") as f:
+            yaml.dump(data, f)
+        print(f"Successfully updated Attribut '{attribute}' in plugin.yaml!")
```
Comment (onkelandy) [inline db_addon/item_attributes_master.py:267]: Attribute (also below)

File: db_addon/item_attributes_master.py [hunk 4]
```diff
@@ -168,27 +205,70 @@ def export_db_addon_data():
...
+        print(f"Attribut '{attribute}' not defined in plugin.yaml")
 
 if __name__ == '__main__':
-    export_db_addon_data()
-    print()
-    print('--------------------------------------------------------------')
-    print()
-    export_for_plugin_yaml()
+    export_item_attributs_py()
+    for attribute in ITEM_ATTRIBUTS:
```
Comment (onkelandy) [inline db_addon/item_attributes_master.py:273]: E missing in attributes (also adjust all occurences)

File: db_addon/item_attributes_master.py (hunk 3 above)
Comment (sisamiwe) [reply db_addon/item_attributes_master.py:267]: Corrected

File: db_addon/item_attributes_master.py (hunk 4 above)
Comment (sisamiwe) [reply db_addon/item_attributes_master.py:273]: Corrected

**Pull request merged**